api/src/main/java/com/capitalone/dashboard/auth/webhook/github/GithubWebHookRequestFilter.java

```diff
@@ -0,0 +1,132 @@
+        if (gitHubWebHookSettings != null) {
+            userAgentExpectedValue = gitHubWebHookSettings.getUserAgent();
+            githubEnterpriseHostExpectedValues = gitHubWebHookSettings.getGithubEnterpriseHosts();
+        }
+
+        if (checkForEmptyValues(userAgent, githubEnterpriseHost, userAgentExpectedValue, githubEnterpriseHostExpectedValues)
+                || !userAgent.contains(userAgentExpectedValue)
+                || !checkGithubEnterpriseHost(githubEnterpriseHost, githubEnterpriseHostExpectedValues)) {
+            authenticated = false;
+            filterChain.doFilter(request, response);
+        } else {
+            authenticated = true;
+            super.doFilter(req, res, filterChain);
+        }
+    }
+
+    private boolean checkForEmptyValues(String userAgent, String githubEnterpriseHost, String userAgentExpectedValue,
+                                List<String> githubEnterpriseHostExpectedValues) {
+        if (HygieiaUtils.checkForEmptyStringValues(userAgent, githubEnterpriseHost, userAgentExpectedValue)
+                || CollectionUtils.isEmpty(githubEnterpriseHostExpectedValues)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) throws AuthenticationException {
+
+        Authentication authentication = githubWebHookAuthService.getAuthentication(httpServletRequest);
+
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+
+        return authentication;
+    }
+
+    @Override
+    protected void successfulAuthentication(HttpServletRequest request, HttpServletResponse response, FilterChain chain,
+                                            Authentication authResult) throws IOException, ServletException {
+        SecurityContextHolder.getContext().setAuthentication(authResult);
+        chain.doFilter(request, response);
+    }
+
+    @Override
+    protected void unsuccessfulAuthentication(HttpServletRequest request, HttpServletResponse response,
+                                              AuthenticationException failed) throws IOException, ServletException {
+
+        response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Github Webhook Authentication Failed");
+    }
+
+    protected boolean checkGithubEnterpriseHost(String githubEnterpriseHost, List<String> githubEnterpriseHostExpectedValues) {
+        String value = Optional.ofNullable(githubEnterpriseHostExpectedValues)
+                .orElseGet(Collections::emptyList).stream()
+                .filter(expectedValue -> githubEnterpriseHost.contains(expectedValue))
+                .findFirst().orElse(null);
+
+        return !StringUtils.isEmpty(value);
+    }
+
+    protected GitHubWebHookSettings parseAsGitHubWebHook(String jsonString) {
+        GitHubWebHookSettings gitHubWebHookSettings = null;
```

Comment:
This method can be eliminated by having GitHubWebHook as its own property file. There is an overhead of repetitive unmarshalling of json from properties file

Reply:
Have implemented a converter to handle webhook properties that gets triggered during the startup, thanks